features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt

```diff
@@ -108,6 +113,22 @@ class MessagesPresenter @AssistedInject constructor(
             mutableStateOf(null)
         }
 
+        var hasDismissedInviteDialog by rememberSaveable {
+            mutableStateOf(false)
+        }
+
+        val inviteProgress = remember { mutableStateOf<Async<Unit>>(Async.Uninitialized) }
+
+        val showReinvitePrompt by remember(
+            hasDismissedInviteDialog,
+            composerState.hasFocus,
+            syncUpdateFlow,
+        ) {
+            derivedStateOf {
+                !hasDismissedInviteDialog && composerState.hasFocus && room.isDirect && room.activeMemberCount == 1L
+            }
+        }
```

Comment:
I kinda want this to be in its own presenter, but that would depend on the composerState from the compose presenter and inter-dependent sub-presenters feels a bit gross. Any ideas? :thinking:

Reply:
I wouldn't mind it too much to have this on a separate presenter, but if we want to avoid that maybe we could move it to a separate `@Composable` function?

Reply:
I'll leave as-is for now. It's only really `showReinvitePrompt` that could move into its own method, and that doesn't make much difference. I might refactor it into a presenter when I have some more time, but that can be in a separate PR.